api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+        );
+
+        $form = new CategoryForm($categoriesApiUrl, $locales, $baseUrl, $temporaryFileApiUrl, $category ?: null);
+        return response()->json($form->getConfig(), Response::HTTP_OK);
+    }
+
+    /**
+     * Delete a category by ID.
+     */
+    public function delete(Request $illuminateRequest): JsonResponse
+    {
+        $categoryId = $illuminateRequest->route('categoryId');
+        $context = $this->getRequest()->getContext();
+
+        $category = Repo::category()->get($categoryId, $context->getId());
+        if (!$category) {
+            return response()->json(__('api.404.resourceNotFound'), Response::HTTP_NOT_FOUND);
+        }
+
+        $image = $category->getImage();
+        Repo::category()->delete($category);
+
+        if ($image) {
+            $publicFileManager = new PublicFileManager();
+            $publicFileManager->removeContextFile($category->getContextId(), $image['uploadName']);
+            $publicFileManager->removeContextFile($category->getContextId(), $image['thumbnailName']);
+        }
+
+        return response()->json([], Response::HTTP_OK);
+    }
+
+    /**
+     * Generate the thumbnail image when creating a category.
+     *
+     * @return array - assoc array with thumbnail details
+     * Example:
+     * ```
+     * [
+     *  'thumbnailName' => (string),
+     *  'thumbnailWidth' => (int),
+     *  'thumbnailHeight' => (int),
+     * ]
+     * ```
+     */
+    private function generateThumbnail(TemporaryFile $temporaryFile, Context $context, $categoryId): array
+    {
+        $temporaryFileManager = new TemporaryFileManager();
+        $imageExtension = $temporaryFileManager->getImageExtension($temporaryFile->getFileType());
+        $sizeArray = getimagesize($temporaryFile->getFilePath());
+        $temporaryFilePath = $temporaryFile->getFilePath();
+
+        // Generate the surrogate images. Used later to create thumbnail
+        $image = match ($imageExtension) {
+            '.jpg' => imagecreatefromjpeg($temporaryFilePath),
+            '.png' => imagecreatefrompng($temporaryFilePath),
+            '.gif' => imagecreatefromgif($temporaryFilePath),
+        };
+
+        $coverThumbnailsMaxWidth = $context->getData('coverThumbnailsMaxWidth');
+        $coverThumbnailsMaxHeight = $context->getData('coverThumbnailsMaxHeight');
+        $thumbnailFilename = $categoryId . '-category-thumbnail' . $imageExtension;
+        $xRatio = min(1, ($coverThumbnailsMaxWidth ? $coverThumbnailsMaxWidth : 100) / $sizeArray[0]);
+        $yRatio = min(1, ($coverThumbnailsMaxHeight ? $coverThumbnailsMaxHeight : 100) / $sizeArray[1]);
+        $ratio = min($xRatio, $yRatio);
+        $thumbnailWidth = round($ratio * $sizeArray[0]);
+        $thumbnailHeight = round($ratio * $sizeArray[1]);
+        $thumbnail = imagecreatetruecolor($thumbnailWidth, $thumbnailHeight);
+        imagecopyresampled($thumbnail, $image, 0, 0, 0, 0, $thumbnailWidth, $thumbnailHeight, $sizeArray[0], $sizeArray[1]);
```

Comment:
Perhaps this should be moved to another place, to be re-used. Or perhaps we could include a library to deal with the cropping/resizing (something for another task).